src/compiler/checker.ts

```diff
@@ -17068,6 +17068,14 @@ namespace ts {
                         return !tp.isThisType && isPartOfTypeNode(node) && maybeTypeParameterReference(node) &&
                             getTypeFromTypeNodeWorker(node as TypeNode) === tp; // use worker because we're looking for === equality
                     case SyntaxKind.TypeQuery:
+                        const entityName = (node as TypeQueryNode).exprName;
+                        const firstIdentifierDeclaration = getSymbolAtLocation(getFirstIdentifier(entityName))?.valueDeclaration;
+                        if (firstIdentifierDeclaration) {
+                            const foundSymbol = resolveName(firstIdentifierDeclaration, tp.symbol.escapedName, SymbolFlags.Type, undefined, undefined, false);
+                            if (tp.symbol && foundSymbol !== tp.symbol && !(node as TypeQueryNode).typeArguments) {

```

Comment:
Also: I'm bailing on this if the type query node has type arguments (e.g. `typeof foo<number>`), but in theory I could recursively call `containsReference` for each type argument and return that instead.